publisher/publisher_test.go

```diff
@@ -247,6 +249,63 @@ func TestUploadArtifacts(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestUploadArtifacts_cantBeRunInParallel(t *testing.T) {
+	t.Skip("flaky on CI")
+
+	schema := []uploadArtifactSchema{
+		{"{app_name}-{arch}-{version}.txt", []string{"amd64"}, []Upload{
+			{
+				Type: "file",
+				Dest: "{arch}/{app_name}/{src}",
+			},
+		}},
+		{"{app_name}-{arch}-{version}.txt", nil, []Upload{
+			{
+				Type: "file",
+				Dest: "{arch}/{app_name}/{src}",
+			},
+		}},
+	}
+
+	dest := t.TempDir()
+	src := t.TempDir()
+	cfg := config{
+		version:              "2.0.0",
+		artifactsDestFolder:  dest,
+		artifactsSrcFolder:   src,
+		uploadSchemaFilePath: "",
+		appName:              "nri-foobar",
+	}
+
+	err := writeDummyFile(path.Join(src, "nri-foobar-amd64-2.0.0.txt"))
+	assert.NoError(t, err)
+
+	ready := make(chan struct{})
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	var err1, err2 error
+	l := lock.NewInMemory()
+	go func() {
+		<-ready
+		err1 = uploadArtifacts(cfg, schema, l)
+		wg.Done()
+	}()
+	go func() {
+		<-ready
+		time.Sleep(1 * time.Millisecond)
```

Comment:
Could it be that this test is flaky because we wait to less here?

Reply:
yep, time based tests are never good, but code will need a revamp in order to test this without incurring into time delays

Reply:
restored